### ISSUE TITLE
Tracking Errors as events

### ISSUE THREAD

**Issue by ArtiBPG** [pg-irc/pathways-frontend#824]
We want to track how often service listings do not load and users get the error messages when they click on "find services near me". 
Error message and scenarios are described in [this issue](https://github.com/pg-irc/pathways-frontend/issues/704) 

Question: Could we do this by treating each error message as an event and tracking through Google Analytics?

### Acceptance Criteria 
- [ ]  On a weekly basis we want to track if users are able to load service listing when they click the "find services near me" button. 
- [ ] If the service listings do not load, we want to know why: no wifi or data, server issue, etc as described in [this issue](https://github.com/pg-irc/pathways-frontend/issues/704)

### Steps for QA if GA approach is chosen: 
1.  Load service listings for 10 topics with wifi and data turned off to expose error screens
2. Test number of events on Google Analytics 

### Steps for QA using Sentry:
1. Create a Sentry account for testing, once created, create a Sentry project

2. Set the following values in your `.env` file:
```
SENTRY_AUTH_TOKEN=your token (see below for how to find this)
SENTRY_DSN=your dsn (see below for how to find this)
SENTRY_ENABLE_IN_DEV=true
SENTRY_DEBUG=true
```  
In Sentry find your auth token at: `/settings/account/api/auth-tokens/` and your DSN at: `/settings/{your-project}/projects/{your-project}/keys/`. You can also just search "dsn" for example from the search bar on the `/settings` page.

3. Start expo with -c

4. Test the errors appear in Sentry (this can sometimes take a minute or two). These are the errors triggered from the topic services screen and how to trigger them:
- "No location": test this by turning off your location
- "Check location": test this on Android (no obvious iOS test we know of) by turning off location high accuracy mode
- "Server error": test this by triggering a bad server response, I go to `src/api/index.ts` and change the endpoint at line ~33 so we're hitting something invalid
- "Offline": this needs to be tested on a physical device by turning off your WIFI

**Comment by bensinjin** [comment]
This should be likely be done with Sentry which we already have in place. TODO:

- Test existing Sentry implementation for this use case

**Comment by rasmus-storjohann-PG** [comment]
QA on android (read device):

- [ ] "No location": test this by turning off your location
- [ ] "Check location": test this on Android (no obvious iOS test we know of) by turning off location high accuracy mode
- [ ] "Server error": test this by triggering a bad server response, I go to src/api/index.ts and change the endpoint at line ~33 so we're hitting something invalid
- [ ] "Offline": this needs to be tested on a physical device by turning off your WIFI

**Comment by tomy-pg** [comment]
Testing on Arrival Advisor Staging version 1.12. 
Device:
Samsung Galaxy S9
Android Version 9 

I received all errors besides "No Location". Similar to @rasmus-storjohann-PG , I get the same error whether my phone location is off or I have high accuracy turned off. Discussed the issue with @bensinjin and this issue will be re-reviewed.

**Comment by bensinjin** [comment]
So I've tested this again, on a Samsung Android device this is what I did:

1. Turn off location services (swipe down from top menu)
2. Open up app in Expo
3. Press "No thanks" when asked to turn on location by the app
4. Observed in Sentry:
<img width="1080" alt="Screen Shot 2020-02-06 at 11 33 09 AM" src="https://user-images.githubusercontent.com/3228323/73971821-7ae02d00-48d4-11ea-8559-4148c0771e10.png">

Maybe we can try these steps once more on one of your devices?
